apps/src/javalab/JavalabCaptchaDialog.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import javalabMsg from '@cdo/javalab/locale';
+import {
+  appendNewlineToConsoleLog,
+  appendOutputLog,
+} from '@cdo/apps/javalab/redux/consoleRedux.ts';
+import {setIsCaptchaDialogOpen} from '@cdo/apps/javalab/redux/javalabRedux.ts';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import ReCaptchaDialog from '@cdo/apps/templates/ReCaptchaDialog';
+
+const BLOG_URL = 'https://support.code.org/hc/en-us/articles/6104078305549';
+
+// exported for tests
+export function UnconnectedJavalabCaptchaDialog({
+  onVerify,
+  onCancel,
+  isCaptchaDialogOpen,
+  setIsCaptchaDialogOpen,
+  appendNewlineToConsoleLog,
+  appendOutputLog,
+  recaptchaSiteKey,
+}) {
+  // return promise for tests
+  const onCaptchaSubmit = token => {
+    return fetch('/dashboardapi/v1/users/me/verify_captcha', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({'g-recaptcha-response': token}),
+    }).then(() => {
+      appendOutputLog(javalabMsg.verificationSuccessful());
+      appendNewlineToConsoleLog();
+      onVerify();
+      setIsCaptchaDialogOpen(false);
+    });
+  };
+
+  const onCaptchaCancel = () => {
+    appendOutputLog(javalabMsg.verificationIncomplete());
+    appendNewlineToConsoleLog();
+    onCancel();
+    setIsCaptchaDialogOpen(false);
+  };
+
+  return (
+    <ReCaptchaDialog
+      submitText={javalabMsg.submit()}
+      isOpen={isCaptchaDialogOpen}
+      handleSubmit={onCaptchaSubmit}
+      handleCancel={onCaptchaCancel}
+      siteKey={recaptchaSiteKey}
+      title={javalabMsg.verificationHeaderMessage()}
+    >
+      <SafeMarkdown
+        markdown={javalabMsg.verificationDialogMessage({blogUrl: BLOG_URL})}
+        openExternalLinksInNewTab
+      />
+    </ReCaptchaDialog>
+  );
+}
+
+UnconnectedJavalabCaptchaDialog.propTypes = {
+  onVerify: PropTypes.func.isRequired,
+  onCancel: PropTypes.func.isRequired,
+  isCaptchaDialogOpen: PropTypes.bool.isRequired,
+  setIsCaptchaDialogOpen: PropTypes.func.isRequired,
+  appendNewlineToConsoleLog: PropTypes.func.isRequired,
+  appendOutputLog: PropTypes.func.isRequired,
+  recaptchaSiteKey: PropTypes.string.isRequired,
+};
+
+export default connect(
```

Comment:
This might make testing more complicated, so not necessary, but you could use [redux hooks](https://react-redux.js.org/api/hooks) in this component and remove all the `connect` and redux props boilerplate